src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs

```diff
@@ -1624,7 +1617,7 @@ private void HandleDirective(DirectiveDescriptor descriptor)
                 }
 
                 Span.ChunkGenerator = new DirectiveTokenChunkGenerator(tokenDescriptor);
-                Output(outputKind, AcceptedCharacters.NonWhiteSpace);
+                Output(SpanKind.Code, AcceptedCharacters.NonWhiteSpace);
```

Comment:
So correct me if I'm wrong, the special case here was that strings were markup and all of the others were code? Seems straightforward.